src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,38 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        final LocalDate dateToDate = LocalDate.parse(dateTo, FORMATTER);
+        final LocalDate dateFromDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateOfLine;
+        int money;
+
+        if (dateFromDate.isAfter(dateToDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder resultString =
+                new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + "\n");
+
+        for (String name: names) {
+            money = 0;
+            resultString.append(name + " - ");
```

Comment:
let's append it at the same place where you call resultString.append(money + "\n");

Reply:
done